addons/account/models/account_journal.py

```diff
@@ -929,3 +929,21 @@ def _get_available_payment_method_lines(self, payment_type):
             return self.inbound_payment_method_line_ids
         else:
             return self.outbound_payment_method_line_ids
+
+    def _is_payment_method_available(self, payment_method_code):
+        """ Check if the payment method is available on this journal. """
+        self.ensure_one()
+        pm_info = self.env['account.payment.method']._get_payment_method_information().get(payment_method_code)
+        if not pm_info:
+            return False
+        currency_ids = pm_info.get('currency_ids')
+        country_id = pm_info.get('country_id')
+        domain = pm_info.get('domain', [('type', 'in', ('bank', 'cash'))])
+        journal_types = domain[0][2]
```

Comment:
How do you know it's the first clause?
Still not perfect, but better:
```suggestion
        journal_types = next(right for left, operator, right in domain if left == 'type' and operator == 'in')
```